mlflow/tracking/client.py

```diff
@@ -2876,6 +2885,7 @@ def print_model_version_info(mv):
             name=name, version=version, description=description
         )
 
+    @deprecated(since="2.9.0")
```

Comment:
It seems that while this decorates the API documentation, it also introduces an extra warning log. So `transition_model_version_stage` will have two warning logs. Please advise.
<img width="1413" alt="Screenshot 2023-11-15 at 12 02 51 PM" src="https://github.com/mlflow/mlflow/assets/66143562/92498e27-587c-4631-8577-c08dc98ec7ba">

Reply:
The deprecated decorator should support handling arbitrary text. Can we just use that one since it auto-applies pydoc notifications as well as the end-use reporting?

you can use the `impact` argument within `@deprecated()` to specify additional information that you would like to have added to the FutureWarning